src/olympia/addons/models.py

```diff
@@ -1260,15 +1260,18 @@ def is_unreviewed(self):
     def is_public(self):
         return self.status == amo.STATUS_PUBLIC and not self.disabled_by_user
 
-    def has_complete_metadata(self):
-        return all(self.get_required_metadata())
-
-    def get_required_metadata(self):
-        if not self.has_listed_versions():
+    def has_complete_metadata(self, for_channel=None):
+        return all(self.get_required_metadata(for_channel=for_channel))
+
+    def get_required_metadata(self, for_channel=None):
+        if for_channel is None:
+            for_channel = (
+                amo.RELEASE_CHANNEL_LISTED if self.has_listed_versions()
+                else amo.RELEASE_CHANNEL_UNLISTED)
+        if for_channel == amo.RELEASE_CHANNEL_UNLISTED:
             # Add-ons with only unlisted versions have no required metadata.
             return []
-        latest_version = self.find_latest_version(
-            channel=amo.RELEASE_CHANNEL_LISTED)
+        latest_version = self.versions.latest()  # Version.from_upload does it.
```

Comment:
`Version.from_upload` does `.latest()` and copies the license information from it.  So, technically, this scenario would work:
1) upload a listed version A (and be required to set the license)
2) upload an unlisted version B (license would be copied from A)
3) delete version A
4) `get_required_metadata` would return truthy values
5) upload a listed version C without specifying a license (or categories, but they're linked to the Addon) and the Addon would be 'complete', because the license for C would be copied from B.